verl/trainer/ppo/ray_trainer.py

```diff
@@ -1108,22 +1108,24 @@ def fit(self):
 
         for epoch in range(self.config.trainer.total_epochs):
             for batch_dict in self.train_dataloader:
+                metrics = {}
+                timing_raw = {}
+
                 do_profile = (
                     self.global_steps in self.config.trainer.profile_steps
                     if self.config.trainer.profile_steps is not None
                     else False
                 )
-                if do_profile:
-                    self.actor_rollout_wg.start_profile(role="e2e", profile_step=self.global_steps)
-                    if self.use_reference_policy:
-                        self.ref_policy_wg.start_profile()
-                    if self.use_critic:
-                        self.critic_wg.start_profile()
-                    if self.use_rm:
-                        self.rm_wg.start_profile()
+                with marked_timer("start_profile", timing_raw):
+                    if do_profile:
+                        self.actor_rollout_wg.start_profile(role="e2e", profile_step=self.global_steps)
```

Comment:
@davidmlw could you wrap these code logic in `self._start_profile()` and `self._stop_profile()` so `fit()` is  shorter?